packages/es-components/src/components/controls/range-selector/RangeSelector.js

```diff
@@ -0,0 +1,180 @@
+    setMinPercentage(initStateMin());
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [currentMinValue, minValue, maxValue]);
+
+  useEffect(() => {
+    setMax(currentMaxValue);
+    setMaxPercentage(initStateMax());
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [currentMaxValue, minValue, maxValue]);
+
+  const changeMin = event => {
+    setMin(Number(event.target.value));
+    setMinPercentage(((event.target.value - minValue) / (maxValue - minValue)) * 100);
+  };
+  const changeMinEnd = event => {
+    setMin(Number(event.target.value));
+    setMinPercentage(((event.target.value - minValue) / (maxValue - minValue)) * 100);
+    onChange(Number(event.target.value), max);
+  };
+  const changeMax = event => {
+    setMax(Number(event.target.value));
+    setMaxPercentage(((event.target.value - minValue) / (maxValue - minValue)) * 100);
+  };
+  const changeMaxEnd = event => {
+    setMax(Number(event.target.value));
+    setMaxPercentage(((event.target.value - minValue) / (maxValue - minValue)) * 100);
+    onChange(min, Number(event.target.value));
+  };
+
+  return (
+    <RangeSelectorBase {...RangeSelectorProps}>
+      <RangeSelectorSlider sliderColor={sliderColor}>
+        <RangeSelectorProgress
+          minPercentage={minPercentage}
+          maxPercentage={maxPercentage}
+          progressColor={progressColor}
+        ></RangeSelectorProgress>
+      </RangeSelectorSlider>
+      <RangeSelectorInputContainer>
+        <RangeSelectorInput
+          type="range"
+          aria-label="min-input"
+          min={minValue}
+          max={maxValue}
+          value={min}
+          onChange={changeMin}
+          onMouseUp={changeMinEnd}
+          progressColor={progressColor}
+          thumbColor={thumbColor}
+        ></RangeSelectorInput>
+        <RangeSelectorInput
+          type="range"
+          aria-label="max-input"
+          min={minValue}
+          max={maxValue}
+          value={max}
+          onChange={changeMax}
+          onMouseUp={changeMaxEnd}
+          progressColor={progressColor}
+          thumbColor={thumbColor}
+        ></RangeSelectorInput>
+      </RangeSelectorInputContainer>
+    </RangeSelectorBase>
+  );
+}
+
+RangeSelector.propTypes = {
+  /** The type attribute for the switch */
+  currentMinValue: PropTypes.number,
+  /** current minimium of range */
+  currentMaxValue: PropTypes.number,
+  /** current maximium of range */
+  minValue: PropTypes.number,
+  /** minimium of range */
+  maxValue: PropTypes.number,
+  /** maximium of range */
+  progressColor: PropTypes.string,
+  /** progress color */
+  sliderColor: PropTypes.string,
+  /** slider color */
+  thumbColor: PropTypes.string,
+  /** slider color */
+  onChange: PropTypes.func
+};
+
+RangeSelector.defaultProps = {
+  currentMinValue: 2500,
+  currentMaxValue: 7500,
+  minValue: 0,
+  maxValue: 10000,
+  progressColor: '#17A2B8',
+  sliderColor: '#ddd',
+  thumbColor: '#17A2B8',
```

Comment:
All these color defaults should be derived from theme values, rather than hard-coded.

Reply:
changed default color to theme color:
progressColor: props => props.theme.colors.success,
  sliderColor: props => props.theme.colors.gray2 ,
  thumbColor: props => props.theme.colors.primary,